db/migrate/20210315163900_update_associations.rb

```diff
@@ -0,0 +1,13 @@
+class UpdateAssociations < ActiveRecord::Migration[5.0]
+  class Spree::Adjustment < ActiveRecord::Base
+    belongs_to :adjustable, polymorphic: true
+    belongs_to :order, class_name: "Spree::Order"
+  end
+
+  def up
+    # Updates any adjustments missing an order association
+    Spree::Adjustment.where(order_id: nil, adjustable_type: "Spree::Order").find_each do |adjustment|
```

Comment:
have you tried

```rb
Spree::Adjustment.where(order_id: nil, adjustable_type: "Spree::Order").update_all("spree_adjustments.order_id = spree_adjustments.adjustable_id")
```

That would be an order of magnitude faster because doesn't require instantiating the adjustment objects.